emails/grunt/replace.js

```diff
@@ -1,16 +1,45 @@
-module.exports = {
-  dist: {
-    overwrite: true,
-    src: ['dist/*.txt'],
-    replacements: [
-      {
-        from: '[[',
-        to: '{{',
-      },
-      {
-        from: ']]',
-        to: '}}',
-      },
-    ],
-  },
+module.exports = function () {
+  'use strict';
+
+  return {
+    txt,
+    comments,
+  };
+};
+
+const txt = {
+  overwrite: true,
+  src: ['dist/*.txt'],
+  replacements: [
+    {
+      from: '[[',
+      to: '{{',
+    },
+    {
+      from: ']]',
+      to: '}}',
+    },
+  ],
+};
+
+/**
+ * Replace all instances of HTML comments with {{ safeHTML "<!-- my comment -->" }}.
+ *
+ * MJML will output <!--[if !mso]><!--> comments which are specific to MS Outlook.
+ *
+ * Go's template/html package will strip all HTML comments and we need them to be preserved
+ * to work with MS Outlook on the Desktop.
+ */
+const HTML_SAFE_FUNC = 'safeHTML';
+const commentBlock = /(\<\!\-\-(?:.|\n|\r)*?-->)/g;
```

Comment:
This regex might be a bit too naive as it will try to match `<!-- foo -->` but comments might still be rendered when there's some other character in the closing HTML comment. One possible character might be `!`, so `<!-- foo --!>` might still be a valid comment, but I'm uncertain if that's an issue in this case.